data/ai/lua/ca_recruit_rushers.lua

```diff
@@ -0,0 +1,32 @@
+-- Make the generic_recruit_engine functions work as external CAs
+
+local dummy_engine = { data = {} }
+local ca_castle_switch = wesnoth.require("ai/lua/ca_castle_switch.lua")
+local params = {
+    score_function = (function() return 300000 end),
+    min_turn_1_recruit = (function() return ca_castle_switch:evaluation({}, dummy_engine.data) > 0 end),
+    leader_takes_village = (function()
+            if ca_castle_switch:evaluation({}, dummy_engine.data) > 0 then
+                local take_village = #(wesnoth.get_villages {
+                    x = dummy_engine.data.leader_target[1],
+                    y = dummy_engine.data.leader_target[2]
+                }) > 0
+                return take_village
+            end
+            return true
```

Comment:
This should be
return not ai.aspects.passive_leader

This is just a bug I noticed because you got me to look this over. If the leader is passive, then ca_castle_switch returns 0, so this path triggers, but the leader won't move, so the correct value should be false. Might be better to do separately thoguh.